### PR TITLE
Trend Micro Tipping Point

### PR DESCRIPTION
**Pull request by mikedgibson**
## Proposed Changes

  - Parser to support custom CEF fields in TippingPoint SMS IPS event logs.
  - Trend Micro TippingPoint Data Connector

### CODE REVIEW

File: DataConnectors/TrendMicroTippingPoint.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            "type": "IsConnectedQuery",
+            "value": [
+                "\nTrendMicroTippingPoint\n\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "dataTypes": [
+        {
+            "name": "CommonSecurityLog (TrendMicroTippingPoint)",
+            "lastDataReceivedQuery": "\nTrendMicroTippingPoint\n\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
```
Comment (v-jayakal) [inline DataConnectors/TrendMicroTippingPoint.json:43]: Please include datatype name, provider name and Appliance name to get appropriate results for connectors. Refer to the comment given for first sample query.

File: DataConnectors/TrendMicroTippingPoint.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        {
+            "description": "Top Destination IP for IPS Events",
+            "query": "\nTrendMicroTippingPoint\n\n            | summarize EventCountTotal = sum(EventCount) by DestinationIP\n            | sort by EventCountTotal desc"
+        }
+    ],
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "\nTrendMicroTippingPoint\n\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
```
Comment (v-jayakal) [inline DataConnectors/TrendMicroTippingPoint.json:36]: Please include datatype name, provider name and Appliance name to get appropriate results for connectors. Refer to the comment given for first sample query.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Review by v-jayakal — APPROVED** [review]

File: Parsers/TrendMicro/TrendMicroTippingPoint [hunk 3]
```diff
@@ -0,0 +1,15 @@
+CommonSecurityLog
```
Comment (v-jayakal) [inline Parsers/TrendMicro/TrendMicroTippingPoint:1]: Could you please include steps required for setting up the parser? Please refer description, usage and references of a parser for a sample here https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/SymantecVIP/SymantecVIP.txt.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Review by oshezaf — CHANGES_REQUESTED** [review]
The parser should use [project-rename](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/projectrenameoperator) rather than extend when only a name change is needed. Naturally, the project-away statement would not need to remove the renamed fields.

File: Parsers/TrendMicro/TrendMicroTippingPoint (hunk 3 above)
Comment (mikedgibson) [reply Parsers/TrendMicro/TrendMicroTippingPoint:22]: Included the steps requested. Thanks for the suggestion.

**Comment by mikedgibson** [comment]
Thanks for the tip on project-rename. I updated the parser with that guidance. For future reference, is it possible to use the value of one field (e.g. DeviceCustomString1Label) as the new field name with project-rename?

**Review by v-jayakal — APPROVED** [review]

**Review by nazang — APPROVED** [review]

File: Parsers/TrendMicro/TrendMicroTippingPoint [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// TippingPoint SMS User Guide: https://docs.trendmicro.com/all/tip/sms/v5.4.0/en-us/sms_5.4.0_ug.pdf
+// 
+CommonSecurityLog
+| where DeviceProduct == "UnityOne"
+| extend DeviceVendor = iff(DeviceVendor == "TippingPoint", "Trend Micro", DeviceVendor)
+| extend DeviceProduct = iff(DeviceProduct == "UnityOne", "TippingPoint", DeviceProduct)
+| parse AdditionalExtensions with "cat=" cat
+| project-rename TippingPointVLAN = DeviceCustomNumber1, TippingPointTaxonomy = DeviceCustomNumber2, TippingPointPacketTrace = DeviceCustomNumber3, TippingPointProfileName = DeviceCustomString1, TippingPointPolicyUUID = DeviceCustomString2, TippingPointSignatureUUID = DeviceCustomString3, TippingPointZoneNames = DeviceCustomString4, TippingPointSMSName = DeviceCustomString5, TippingPointFilterMessageParms = DeviceCustomString6
+| project-away DeviceCustomIPv6Address1Label, DeviceCustomIPv6Address2Label, DeviceCustomIPv6Address3Label, DeviceCustomNumber1Label, DeviceCustomNumber2Label, DeviceCustomNumber3Label, DeviceCustomString1Label, DeviceCustomString2Label, DeviceCustomString3Label, DeviceCustomString4Label, DeviceCustomString5Label, DeviceCustomString6Label
```
Comment (nazang) [inline Parsers/TrendMicro/TrendMicroTippingPoint:28]: @mikedgibson, according to Offir, the project-away clause is not needed for rename columns, do you try it and believe project-away is still needed for the parser?

**Pull request merged**